awx/main/tests/functional/models/test_notifications.py

```diff
@@ -105,7 +105,10 @@ def check_structure(expected_structure, obj):
                         assert isinstance(obj[key], dict)
                         check_structure(expected_structure[key], obj[key])
                     else:
-                        assert isinstance(obj[key], expected_structure[key])
+                        if key == 'job_explanation':
+                            assert isinstance(str(obj[key]), expected_structure[key])
```

Comment:
I don't love this, but I don't see a better way.  Without this, the functional context tests fail when checking the type of the `obj.job_explanation`:

```
 +  where False = isinstance('', <class 'str'>)
```

This is because Django creates a proxy object to keep track of this string:

```
>>> type(obj.job_explanation)
<class 'str'>
>>> type(_(obj.job_explanation))
<class 'django.utils.functional.lazy.<locals>.__proxy__'>
```

The best way would be to import that `__proxy__` object and check against that, but I don't seem to be able to import it
```
from django.utils.functional import __proxy__
```

Reply:
Given that this is a test, I think this is fine.